chia/full_node/coin_store.py

```diff
@@ -179,6 +179,36 @@ async def get_coin_record(self, coin_name: bytes32) -> Optional[CoinRecord]:
                     return record
         return None
 
+    async def get_coin_records(self, names: List[bytes32]) -> List[CoinRecord]:
+        if len(names) == 0:
+            return []
+
+        coins = set()
+        async with self.db_wrapper.read_db() as conn:
+            cursors: List[Awaitable[Cursor]] = []
+            for names_chunk in chunks(names, MAX_SQLITE_PARAMETERS):
+                names_db: Tuple[Any, ...]
+                if self.db_wrapper.db_version == 2:
+                    names_db = tuple(names_chunk)
+                else:
+                    names_db = tuple([n.hex() for n in names_chunk])
+                cursors.append(
+                    conn.execute(
+                        f"SELECT confirmed_index, spent_index, coinbase, puzzle_hash, "
+                        f"coin_parent, amount, timestamp FROM coin_record "
+                        f'WHERE coin_name in ({"?," * (len(names_db) - 1)}?) ',
```

Comment:
the -1 is to compensate for the last `?` at the end, that doesn't have a `,` after it. I agree that `join()` is a more elegant solution to that

Reply:
how about:

```
",".join(["?"] * len(names_db))
```
?

Reply:
Oh I see, I didn't notice the `?` after the `}`.

I guess I actually meant `", ".join("?" for _ in range(len(names_db)))` (ie. I forgot `range`).

Your way works too of course. I've never seen it written that way. Your way knows the string length in advance, which means python could conceivably avoid doing `realloc` (although I don't know for a fact it does do that) at the cost of allocating memory for the list. My way likely has to allocate memory for the iterator, since its current state has to be saved somewhere. In the grand scheme of things, it's all just nit-picky and pretty irrelevant since it's not a very tight loop.